src/order-book/api.ts

```diff
@@ -0,0 +1,180 @@
+  OrderCancellation,
+  OrderCreation,
+  OrderPostError,
+  OrderQuoteRequest,
+  OrderQuoteResponse,
+  Trade,
+  TransactionHash,
+  UID,
+} from './generated'
+import { CowError } from '../common/cow-error'
+import { SupportedChainId } from '../common/chains'
+import {
+  ApiContext,
+  CowEnv,
+  DEFAULT_COW_API_CONTEXT,
+  EnvConfigs,
+  PartialApiContext,
+  PROD_CONFIG,
+  STAGING_CONFIG,
+} from '../common/configs'
+import { transformOrder } from './transformOrder'
+import { EnrichedOrder } from './types'
+import { ApiRequestOptions } from './generated/core/ApiRequestOptions'
+import { request as __request } from './generated/core/request'
+
+class FetchHttpRequest extends BaseHttpRequest {
+  constructor(config: OpenAPIConfig) {
+    super(config)
+  }
+
+  /**
+   * Request method
+   * @param options The request options from the service
+   * @returns CancelablePromise<T>
+   * @throws ApiError
+   */
+  public override request<T>(options: ApiRequestOptions): CancelablePromise<T> {
+    return __request(this.config, {
+      ...options,
+      headers: {
+        ...options.headers,
+        'Content-Type': 'application/json',
+      },
+    })
+  }
+}
+
+export class OrderBookApi {
+  public context: ApiContext
+
+  private servicePerNetwork: Record<SupportedChainId, OrderBookClient | null> = {
+    [SupportedChainId.MAINNET]: null,
+    [SupportedChainId.GOERLI]: null,
+    [SupportedChainId.GNOSIS_CHAIN]: null,
+  }
+
+  constructor(context: PartialApiContext = {}) {
+    this.context = { ...DEFAULT_COW_API_CONTEXT, ...context }
+  }
+
+  getTrades(
+    { owner, orderId }: { owner?: Address; orderId?: UID },
+    contextOverride: PartialApiContext = {}
+  ): CancelablePromise<Array<Trade>> {
+    if (owner && orderId) {
+      return new CancelablePromise((_, reject) => {
+        reject(new CowError('Cannot specify both owner and orderId'))
+      })
+    }
+
+    return this.getServiceForNetwork(contextOverride).getApiV1Trades(owner, orderId)
+  }
+
+  getOrders(
+    {
+      owner,
+      offset = 0,
+      limit = 1000,
+    }: {
+      owner: Address
+      offset?: number
+      limit?: number
+    },
+    contextOverride: PartialApiContext = {}
+  ): Promise<Array<EnrichedOrder>> {
+    return this.getServiceForNetwork(contextOverride)
+      .getApiV1AccountOrders(owner, offset, limit)
+      .then((orders) => {
+        return orders.map(transformOrder)
+      })
+  }
+
+  getTxOrders(txHash: TransactionHash, contextOverride: PartialApiContext = {}): Promise<Array<EnrichedOrder>> {
+    return this.getServiceForNetwork(contextOverride)
+      .getApiV1TransactionsOrders(txHash)
+      .then((orders) => {
+        return orders.map(transformOrder)
+      })
+  }
+
+  getOrder(uid: UID, contextOverride: PartialApiContext = {}): Promise<EnrichedOrder> {
+    return this.getServiceForNetwork(contextOverride)
+      .getApiV1Orders(uid)
+      .then((order) => {
+        return transformOrder(order)
+      })
+  }
+
```

Comment:
Weren't you going to add a `getOrderMultiEnv`?

Reply:
Thank you, I forgot about it :)
Added